create_heatmaps_job.py

```diff
@@ -1,71 +1,347 @@
-    parser.add_argument('--start', type=int, help='metadata/lcdata files index to start processing at')
-    parser.add_argument('--end', type=int, help='metadata/lcdata files index to stop processing at')
+
+    parser.add_argument('--config_path', type=str, 
+                        help='absolute or relative path to your yml config file, i.e. "/user/files/create_heatmaps_config.yml"')
+
+    # input explicit start and end jobid (original feature)
+    parser.add_argument('--start', type=int, default=None,
+                        help='metadata/lcdata files index to start processing at')
+    parser.add_argument('--end', type=int, default=None,
+                        help='metadata/lcdata files index to stop processing at')
+
+    # ... OR ...
+
+    # input slurm_id and nslurm_tot and internally figure out jobids (RK - mar 2024)
+    # This input option requires no user knowledge of how many FITS/PHOT files.
+    parser.add_argument('--slurm_id', type=int, default=None,
+                        help='jobid from 0 to njobtot-1')
+    parser.add_argument('--nslurm_tot', type=int, default=None,
+                        help='total number or slurm jobs')
+
+    # optional redirect of heatmaps subdir
+    parser.add_argument('--heatmaps_subdir', type=str, default=HEATMAPS_SUBDIR_DEFAULT,
+                        help='alternate heatmaps subdir output')
+
     args = parser.parse_args()
+
+    # define and append string to identify job in logging (not used for computation)
+    if args.start is not None:
+        args.jobid_string = f"jobid_{args.start}-{args.end}"  
+    else:
+        args.jobid_string = f"slurm_id{args.slurm_id:02d}"
+
     return args
 
-def load_config(config_path):
-    with open(config_path, "r") as cfgfile:
-        config = yaml.load(cfgfile, Loader=yaml.Loader)
-    # expand env vars
-    config['lcdata_paths'] = [os.path.expandvars(path) for path in config['lcdata_paths']]
-    config['metadata_paths'] = [os.path.expandvars(path) for path in config['metadata_paths']]
-    config['heatmaps_path'] = os.path.expandvars(config['heatmaps_path'])
-    config['output_path'] = os.path.expandvars(config['output_path'])
-    config['trained_model'] = os.path.expandvars(str(config['trained_model'])) if 'trained_model' in config and os.path.exists(str(config['trained_model'])) else None
+
+def load_config(args):
+
+    config_path = args.config_path
+
+    key_expandvar_list = [ 'input_data_paths', 'lcdata_paths', 'metadata_paths', 
+                           'heatmaps_path', 'output_path', 
+                           'heatmaps_logfile', 'heatmaps_donefile'  ]
+
+    config = util.load_config_expandvars(config_path, key_expandvar_list)
+    
+    # if sim data folders are provided, read list file(s) and set internal array
+    # for each PHOT.FITS and *HEAD.FITS file
+    load_lcdata_metadata(config)
+
+    key = 'trained_model'
```

Comment:
are these `key` variables necessary in this line and below lines? why not just `config['trained_model'] = ...`

Reply:
technically key variable is not needed. However, it is used four times in code below, so if logic ever changes (e.g., looping over multiple key names, or extracting key name from somewhere else), it reduces chance of mistake.